lms/djangoapps/courseware/module_render.py

```diff
@@ -807,6 +792,27 @@ def rebind_noauth_module_to_user(module, real_user):
             'fs': FSService(),
             'field-data': field_data,
             'user': DjangoXBlockUserService(user, user_is_staff=user_is_staff),
+            'render': RenderService(
+                render_template=render_to_string,
+                # TODO (cpennington): This should be removed when all html from
+                # a module is coming through get_html and is therefore covered
+                # by the replace_static_urls code below
```

Comment:
Hmm, it does not look like `render_to_string()` is doing the [processing](https://github.com/edx/edx-platform/blob/master/common/djangoapps/static_replace/__init__.py#L150) that is done in the next two methods? Maybe this comment is outdated?

Reply:
From what I can see, [`runtime.replace_*urls` is only being used by the CAPA ProblemBlock](https://github.com/edx/edx-platform/search?q=runtime.replace_urls); the rest of the XBlocks use them via the [block_wrappers](https://github.com/edx/edx-platform/blob/78c6be8beeb49ae6edb132147b6ecdaa21664f9c/lms/djangoapps/courseware/module_render.py#L705-L726) passed from the LMS.

So I think these could be removed from the ModuleSystem entirely, along with several other properties used only by CAPA modules (e.g. `cache`, `can_execute_unsafe_code`, `xqueue`, `get_python_lib_zip`). I'll do it as part of a separate task, but maybe these could be moved to the LoncapaSystem?